File: fundamentais/javascript/conversao_base.js
// Algoritmo Fundamental Nº 8
// Descrição : Conversão de base
// Problema  : Converter um número inteiro decimal para a
//             sua representação binária correspondente.
// Aplicações: Interpretação de dados e instruções armazenados em computador.

var n = prompt("Número inteiro decimal:");

var q = n;
var ndigit = 0;
var r;
var binario = [];

do {
	r = q % 2;
    binario[ndigit] = r;
    ndigit = ndigit + 1;
    q = Math.trunc(q / 2);
} while (q != 0);

document.write("Número binário: ");

for (var i = ndigit-1; i >= 0; i--) {
    document.write(binario[i]);
}

File: fundamentais/java/ConversaoBase.java
import java.util.Scanner;

public class ConversaoBase {

	public static void main(String[] args) {
		Scanner dado = new Scanner(System.in);
		
		System.out.print("Informe um número inteiro decimal: ");
		int numero = dado.nextInt();

		int q = numero;
		int resto[] = new int[100];
		int i = 0;

		while (q > 0)
		{
		    resto[i] = q % 2;
		    i++;
		    q = (int) Math.floor(q/2);
		}

		System.out.println("Número na base decimal: " + numero);
		System.out.print("Número na base binária: ");

		for (int j = i; j >=0; j--) {
			System.out.print(resto[j]);
		}
	}

}